packages/dev/hardhat.config.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+import '@nomiclabs/hardhat-truffle5'
+import '@nomiclabs/hardhat-web3'
+import 'hardhat-deploy'
+import '@nomiclabs/hardhat-ethers'
+import '@nomiclabs/hardhat-etherscan'
+
+import fs from 'fs'
+import { HardhatUserConfig } from 'hardhat/config'
+import { NetworkUserConfig } from 'hardhat/src/types/config'
+import path from 'path'
+import chalk from 'chalk'
+
+const mnemonicFileName = process.env.MNEMONIC_FILE
+let mnemonic = 'test '.repeat(11) + 'junk'
+if (mnemonicFileName != null && fs.existsSync(mnemonicFileName)) {
+  mnemonic = fs.readFileSync(mnemonicFileName, 'ascii')
+}
+
+function getNetwork (url: string): NetworkUserConfig {
+  return {
+    url,
+    accounts: { mnemonic }
+  }
+}
+
+const infuraUrl = (name: string): string => `https://${name}.infura.io/v3/${process.env.INFURA_ID}`
+
+function getInfuraNetwork (name: string): NetworkUserConfig {
+  return getNetwork(infuraUrl(name))
+}
+
+const CONTRACTS_LINK = 'contracts-link'
+
+if (!fs.existsSync(path.join(CONTRACTS_LINK, 'RelayHub.sol'))) {
+  console.log('== creating symlink', chalk.yellow(CONTRACTS_LINK), 'for contracts')
+  fs.symlinkSync('../contracts/solpp', CONTRACTS_LINK)
```

Comment:
Is this really necessary? I don't like this touching the file system from the config file. It can be executed by the engine dozens of times.